### PR TITLE
able to build and start dockerized rpc adaptor

### PR DESCRIPTION
**Pull request by shunjizhan**
## Change
Added Dockerfile to build adn run `eth-rpc-adaptor` service.

Note that this uses `yarn` for itself, instead of `rush` for monorepo, so it needs a slightly different `package.json`, and a `yarn.lock` just for docker (rush won't need to use it)

## Test
- able to start a dockerized `eth-rpc-adaptor` service, and connected to the latest public mandala testnet.
- tested a couple `eth_xxx` and they worked.

## TODO
currently any `eth_xxx` that interacts with evm-subql won't work, we will need to add support for it after we have a public evm-subql service.

### CODE REVIEW

**Review by xlc — COMMENTED** [review]
better to have ./docker/package.json and ./docker/yarn.lock so it is easier to update them and the IDE understand the types

File: eth-rpc-adapter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM node:14-alpine
+
+ARG mandala_url=wss://node-6870830370282213376.rz.onfinality.io/ws?apikey=0f273197-e4d5-45e2-b23e-03b015cb7000
```
Comment (xlc) [inline eth-rpc-adapter/Dockerfile:3]: No need to inject this at build time. We can configure environment variable at deployment time.

**Comment by shunjizhan** [comment]
> better to have ./docker/package.json and ./docker/yarn.lock so it is easier to update them and the IDE understand the types

moved

> No need to inject this at build time. We can configure environment variable at deployment time.

removed, now uses local `.env` file, same as running locally

File: eth-rpc-adapter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM node:14-alpine
+
```
Comment (tj4laminar) [inline eth-rpc-adapter/Dockerfile:2]: is it current latest version of node-alpine image ?

File: eth-rpc-adapter/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ENV NODE_ENV=production
+
```
Comment (tj4laminar) [inline eth-rpc-adapter/Dockerfile:4]: is it required field or just get as it was in example code which shared by Bryan ?

File: eth-rpc-adapter/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+COPY . .
+COPY docker/package.json ./package.json
+COPY docker/yarn.lock ./yarn.lock
+
+RUN yarn
+RUN [ "yarn", "build" ]
+
+EXPOSE 8545
+
```
Comment (tj4laminar) [inline eth-rpc-adapter/Dockerfile:15]: is it kind of incoming port or just connecting to external world ?

File: eth-rpc-adapter/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /usr/src/app
+
+COPY . .
+COPY docker/package.json ./package.json
+COPY docker/yarn.lock ./yarn.lock
+
+RUN yarn
+RUN [ "yarn", "build" ]
+
```
Comment (tj4laminar) [inline eth-rpc-adapter/Dockerfile:13]: let run this container as non-root user

File: eth-rpc-adapter/Dockerfile (hunk 3 above)
Comment (xlc) [reply eth-rpc-adapter/Dockerfile:4]: we want to have this for prod image

File: eth-rpc-adapter/Dockerfile (hunk 4 above)
Comment (xlc) [reply eth-rpc-adapter/Dockerfile:15]: this will be exposed to external world

File: eth-rpc-adapter/Dockerfile (hunk 2 above)
Comment (shunjizhan) [reply eth-rpc-adapter/Dockerfile:2]: latest version is 17 but we use 14 across `bodhi.js` project

File: eth-rpc-adapter/Dockerfile (hunk 5 above)
Comment (shunjizhan) [reply eth-rpc-adapter/Dockerfile:13]: now use user 'node'

**Review by tj4laminar — APPROVED** [review]
looks good

**Pull request merged**